### DISCUSSION TITLE
When I use router.refresh() it redirects to https

### DISCUSSION THREAD

**Discussion by bibble235** [vercel/next.js#57863] [Help]
### Summary

Hi,

I want to refresh a page when I add a record. To do this I use

`router.refresh()`

This seems to redirect to the https://localhost:3000/blah/blah rather than http://localhost:3000/blah/blah

I am running 
Ubuntu 23.04
Chrome Version 118.0.5993.117
NextJS 14.0.1

To get an app to work I have to do the HSTS thing
go to chrome://net-internals/#hsts 
delete localhost to stop redirecting to https in the first place.

The backend API runs on https://localhost:1701

Any help greatly appreciated

**Comment by icyJoseph** [comment]
Hi,

DO you observe this with other browsers? I wonder if Chrome has started to enforce automatic HTTPS upgrades.

**Comment by bibble235** [comment]
Tried with FF but Cors issues. Will feedback tonight.

**Comment by bibble235** [comment]
The cors issue was just me needing to import cert. (do not use FF usually). Total expected behavior.

With FF the router.refresh() no longer re-direct to https. 

It does not refresh the content on the page but I can see on the server log it does re-fetch the table data being added to. (this might be a separate issue 11pm here)

**Reply by bibble235** [reply]
So this does appear to be a problem.

`
            startTransition(async () => {
                await deleteFileRecords(rowsSelected, apiTokenDetails)
            })

            console.log('Finished Awaiting Delete')
            setErrorMessage(() => `${rowsSelected.length} Files Deleted`)

            router.refresh()
`

On Firefox this code behaves as expected. The server action runs and refreshes correctly

On Chrome we are redirected to the https (307) where FF is success (200) and get ERR_SSL_PROTOCOL_ERROR as a consequence.

**Comment by bibble235** [comment]
To work around this you can either
a) Use Firefox
b) Run Nextjs with next dev --experimental-https

**Reply by icyJoseph** [reply]
But is this an issue because of the Backend you have, that's using https?

**Comment by bibble235** [comment]
I would say, it could be. Maybe the accessing of https://localhost:3016 where the backend runs, removes the delete from chrome://net-internals/#hsts for localhost and subsequently causes this on the refresh. The option clearly is not limited to a particular port.

End work around was to move to https with 
next dev \
--experimental-https \
--experimental-https-cert='/etc/localcerts/192.xyz.1.123/server.crt' \  
--experimental-https-key='/etc/localcerts/192.xyz.1.123/server.key'

No issues now on either FF or Chrome

I would be nice to understand how to prevent this though
